### PR TITLE
DEV ONLY // fix bug in Table selection when filter is applied

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated Table story](https://deploy-preview-565--neo-react-library-storybook.netlify.app/?path=/story/components-table--pre-selected-rows)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [ ] ~tagged `@avaya-dux/dux-design` if any visual changes have occurred~
- [x] tagged `@avaya-dux/dux-devs`

Bhavesh noted that the table body's "select"/"clear" all, the header's dropdown menu items, and the header's checkbox were not return all selected rows when clicked if a filter is applied to the table. This PR fixes those issues so all selected items are always returned. 

`@avaya-dux/dux-devs`: review code, story functionality, and that appropriate tests have been written 

- note that I've only written one unit test for this as all the methods used in the example GIF call the same helper methods. Thus why the only functional change is in `helpers.ts`. And that change is simply spreading the previously selected items.

**EXAMPLE FUNCTIONALITY GIF:**

## Summary by CodeRabbit

- **New Features**
  - Added new stories to the Table component in Storybook, including pre-selected rows, second page, and disabled rows.
  - Enhanced user interaction with new custom actions and a filter drawer for the Table component.
  - Updated the version of the project to 1.3.8.

- **Bug Fixes**
  - Improved row selection functionality to ensure accurate selection state when data is filtered.

- **Tests**
  - Introduced a new test case for verifying row selection accuracy during data filtering.

### CODE REVIEW

File: src/components/Table/helpers/helpers.ts [hunk 1]
```diff
@@ -50,14 +50,24 @@ export const setTableRowsSelected = <T extends Record<string, unknown>>(
...
 	enabledTableRowsIds.forEach((id) => toggleRowSelected(id, selected));
 
-	handleRowToggled?.(selected ? enabledTableRowsIds : []);
+	const selectedRowIdsArray = Object.keys(selectedRowIds);
+
+	const adjustedSelectedRowIds = selected
+		? [...selectedRowIdsArray, ...enabledTableRowsIds]
+		: selectedRowIdsArray.filter((id) => !enabledTableRowsIds.includes(id));
+
+	handleRowToggled?.(adjustedSelectedRowIds);
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Table/helpers/helpers.ts]: updated per recommendation, please confirm that I've applied the fix as you have recommended

File: src/components/Table/Table.test.jsx [hunk 2]
```diff
@@ -512,6 +512,27 @@ describe("Table", () => {
...
+			expect(codeElement).toHaveTextContent(
+				'["10","40","20","10","30","40","50","60","70","80","90","100"]',
+			);
+
+			const deselectAllItemsButton = screen.getByText(
+				FilledFields.translations.header.clearAll,
+			);
+			await user.click(deselectAllItemsButton);
+			expect(codeElement).toHaveTextContent("[]");
+		});
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Table/Table.test.jsx]: updated, resolving

File: src/components/Table/helpers/helpers.ts [hunk 3]
```diff
@@ -72,12 +82,22 @@
...
 	enabledPageRowsIds.forEach((id) => toggleRowSelected(id, selected));
 
-	handleRowToggled?.(selected ? enabledPageRowsIds : []);
+	const selectedRowIdsArray = Object.keys(selectedRowIds);
+
+	const adjustedSelectedRowIds = selected
+		? [...selectedRowIdsArray, ...enabledPageRowsIds]
+		: selectedRowIdsArray.filter((id) => !enabledPageRowsIds.includes(id));
+
+	handleRowToggled?.(adjustedSelectedRowIds);
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Table/helpers/helpers.ts]: included in last update, please confirm that this is working as expected

File: src/components/Table/helpers/helpers.ts [hunk 4]
```diff
@@ -50,14 +50,24 @@
...
 	enabledTableRowsIds.forEach((id) => toggleRowSelected(id, selected));
 
-	handleRowToggled?.(selected ? enabledTableRowsIds : []);
+	const selectedRowIdsArray = Object.keys(selectedRowIds);
+
+	const adjustedSelectedRowIds = selected
+		? [...selectedRowIdsArray, ...enabledTableRowsIds]
+		: selectedRowIdsArray.filter((id) => !enabledTableRowsIds.includes(id));
+
+	handleRowToggled?.(adjustedSelectedRowIds);
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Table/helpers/helpers.ts]: included in last update, please confirm that this is working as expected

File: src/components/Table/helpers/helpers.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Table/helpers/helpers.ts]: 🔥 resolving

File: src/components/Table/helpers/helpers.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Table/helpers/helpers.ts]: 🔥 resolving

File: src/components/Table/helpers/helpers.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Table/helpers/helpers.ts]: 🔥 resolving

**Review by enrique-prado — APPROVED** [review]
Nice job.

**Pull request merged**